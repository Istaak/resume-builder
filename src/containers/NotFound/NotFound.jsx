// Third-party dependencies
import React from 'react';
import { useSelector } from 'react-redux';

// Material UI
import Button from '@material-ui/core/Button';
import { selectors } from 'constants/strings.store';

// Styles
import './NotFound.scss';

const NotFound = () => {
  const languageConstants = useSelector(selectors.constants);
  const constants = languageConstants.pages.notFound;
  const fullLogo = `${process.env.PUBLIC_URL}/assets/img/full_logo.png`;

  return (
    <section className="not-found">
      <div className="imagewrapper">
        4
        <img src={fullLogo} alt="leviahack_logo" />
        4
      </div>

      <div className="message">
        <h1>{ constants.title }</h1>
        <p>{ constants.description }</p>
        <Button variant="contained" color="primary">
          { constants.button }
        </Button>
      </div>
    </section>
  );
};

export default NotFound;
