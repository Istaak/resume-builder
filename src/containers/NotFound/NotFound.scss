@import '../../styles/abstracts/variables';
@import '../../styles/abstracts/mixins';

.not-found {
  width: 100vw;
  height: 100vh;
  margin: 0;
  padding: 30px;
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  align-content: center;

  &::before {
    content: '';
    background: var(--theme-secondary-color);
    position: absolute;
    width: 200px;
    height: 10px;
    top: 80px;
    left: -5px;
    border-radius: 5px;
  }

  &::after {
    content: '';
    background: var(--theme-primary-color);
    position: absolute;
    width: 300px;
    height: 10px;
    top: 100px;
    left: -5px;
    border-radius: 5px;
  }

  > .imagewrapper {
    text-align: center;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 20vw;

    img {
      width: 150px;
      margin: 0 20px;
    }
  }

  > .message {
    text-align: center;
    width: 100%;
    padding: 0 30px;

    h1 {
      font-size: 35px;
    }

    p {
      font-size: 20px;
      color: map-get($palette, 'dusty-gray');
    }

    button {
      background-color: var(--theme-primary-color);
      margin-top: 30px;
    }
  }
}

@include media-breakpoint-up(lg) {
  .not-found {
    > .imagewrapper,
    > .message {
      flex-basis: 50%;
    }

    > .imagewrapper {
      font-size: 15vw;

      img {
        width: 250px;
      }
    }
  }
}
