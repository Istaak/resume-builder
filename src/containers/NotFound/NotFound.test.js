// Third-party dependencies
import React from 'react';
import ReactTestRender from 'react-test-renderer';

// Own constants/actions/components
import NotFound from './NotFound';

describe('NotFound: ', () => {
  test('should render correctly', () => {
    const component = ReactTestRender.create(<NotFound />);
    const render = component.toJSON();
    expect(render).toMatchSnapshot();
  });
});
