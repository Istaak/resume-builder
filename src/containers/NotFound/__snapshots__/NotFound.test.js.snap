// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound:  should render correctly 1`] = `
<section
  className="not-found"
>
  <div
    className="imagewrapper"
  >
    4
    <img
      alt="leviahack_logo"
      src="/assets/img/full_logo.png"
    />
    4
  </div>
  <div
    className="message"
  >
    <h1>
      Ops! No se encontr칩 la p치gina
    </h1>
    <p>
      Desafortunadamente la p치gina que est치s buscando se ha movido o eliminado
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Ir a Inicio
      </span>
    </button>
  </div>
</section>
`;
